client/task_runner.go

```diff
@@ -1474,6 +1463,30 @@ func (r *TaskRunner) Destroy(event *structs.TaskEvent) {
 	close(r.destroyCh)
 }
 
+// getCreatedResources returns the resources created by drivers. It will never
+// return nil.
+func (r *TaskRunner) getCreatedResources() *driver.CreatedResources {
+	r.createdResourcesLock.Lock()
+	if r.createdResources == nil {
+		r.createdResources = driver.NewCreatedResources()
+	}
+	cr := r.createdResources.Copy()
+	r.createdResourcesLock.Unlock()
+
+	return cr
+}
+
+// setCreatedResources updates the resources created by drivers. If passed nil
+// it will set createdResources to an initialized struct.
+func (r *TaskRunner) setCreatedResources(cr *driver.CreatedResources) {
+	if cr == nil {
+		cr = driver.NewCreatedResources()
```

Comment:
Well since this is for safety I figured might as well always set it.

I could have avoided the Copy call on this path too, but it shouldn't cost much or be called often.

Honestly I've botched the seemingly trivial code around this field so many times now that I just wanted these methods to be as succinct and robust as possible.